cfbs/cfbs_config.py

```diff
@@ -485,3 +485,30 @@ def _input_list(input_data):
                 definition["response"] = _input_list(definition)
             else:
                 user_error("Unsupported input type '%s'" % definition["type"])
+
+    def _get_all_module_names(self, search_in=("build", "provides", "index")):
+        modules = []
+
+        if "build" in search_in and "build" in self:
+            modules.extend((x["name"] for x in self["build"]))
+        if "provides" in search_in and "provides" in self:
+            modules.extend(self["provides"].keys())
+        if "index" in search_in:
+            modules.extend(self.index.keys())
+
+        return modules
+
+    def can_reach_dependency(self, name, search_in=("build", "provides", "index")):
+        return name in self._get_all_module_names()
+
+    def find_module(self, name, search_in=("build", "provides", "index")):
+        if "build" in search_in and "build" in self:
+            for module in (x for x in self["build"]):
```

Comment:
Is not this the same? 🤔 
```suggestion
            for module in self["build"]:
```

Reply:
👍 Yea, for a list it is, for an array it would not be.